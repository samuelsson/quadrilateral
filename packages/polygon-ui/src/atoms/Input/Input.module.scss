@import '../../styles/abstracts/colors';

[data-theme='light'] {
  --Input-background-color: #{$color-white};
  --Input-border-color: #{$color-blue-gray-200};
  --Input-box-shadow: 0 4px 15px #{$color-gray-300};
}

[data-theme='dark'] {
  --Input-background-color: #{$color-blue-gray-700};
  --Input-border-color: #{$color-blue-gray-900};
  --Input-box-shadow: none;
}

.Input {
  height: 48px;
  padding: 0 16px;
  border-radius: 8px;
  border: 2px solid;
  border-color: var(--Input-border-color);
  box-sizing: border-box;
  background-color: var(--Input-background-color);
  box-shadow: var(--Input-box-shadow);
  color: inherit;
  font-size: 1rem;
  transition-property: box-shadow, border, opacity;
  transition-timing-function: linear;
  transition-duration: 0.2s;

  &:focus:not(:disabled),
  &:active:not(:disabled) {
    border-color: $color-blue;
  }

  &:disabled {
    opacity: 0.5;
    cursor: not-allowed;
  }
}

.Error {
  border-color: $color-red;
}
