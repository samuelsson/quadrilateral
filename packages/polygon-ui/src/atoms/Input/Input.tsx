import React from 'react';
import cn from 'classnames';
import styles from './Input.module.scss';

export interface InputProps {
  name: string;
  type: string;
  disabled?: boolean;
  error?: boolean;
  placeholder?: string;
}

const Input = ({
  name,
  type,
  disabled,
  error,
  placeholder,
}: InputProps): JSX.Element => (
  <input
    name={name}
    type={type}
    disabled={disabled}
    placeholder={placeholder}
    className={cn(styles.Input, error && styles.Error)}
  />
);

export default Input;
