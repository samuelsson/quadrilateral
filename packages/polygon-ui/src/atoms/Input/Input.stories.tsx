import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { boolean, radios, text } from '@storybook/addon-knobs';

import Input from './Input';

storiesOf('Input', module).add(
  'Default',
  (): JSX.Element => (
    <Input
      type={radios('Type', { Text: 'text', Password: 'password' }, 'text')}
      name="input"
      placeholder={text('Placeholder', '')}
      disabled={boolean('Disabled', false)}
      error={boolean('Error', false)}
    />
  )
);
