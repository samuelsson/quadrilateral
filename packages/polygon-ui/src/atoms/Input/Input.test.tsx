import * as React from 'react';
import { render } from '@testing-library/react';
import Input from './Input';

describe('Input', (): void => {
  it('renders with no errors', (): void => {
    const { asFragment } = render(<Input name="input" type="text" />);

    expect(asFragment()).toMatchSnapshot();
  });
});
