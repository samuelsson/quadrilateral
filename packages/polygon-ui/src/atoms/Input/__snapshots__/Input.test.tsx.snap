// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders with no errors 1`] = `
<DocumentFragment>
  <input
    class="Input"
    name="input"
    type="text"
  />
</DocumentFragment>
`;
